### PR TITLE
chore: Update datafusion and use released version of arrow crates

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
Pull in the most recent DataFusion dependencies as well as arrow

Hopefully from now on we will only have to manage DataFusion upgrades and cargo can do the incremental upgrades

# Changes
1. Update version of datafusion to latest gitsha
2. Update version of arrow to "4.0.0" (which really means 4.x)
3. Run `cargo update`
4. Update parquet reading code to use changed APIs

### CODE REVIEW

File: arrow_util/Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ description = "Apache Arrow utilities"
 
 [dependencies]
 
-arrow = { path = "../arrow" }
```
Comment (alamb) [inline arrow_util/Cargo.toml:10]: Even though this says "4.0.0" the way cargo works it really means the latest "4.x"

You can see that we are using `4.1.0` here via a command such as:
```
alamb@ip-10-0-0-124 influxdb_iox % cargo tree | grep arrow
cargo tree | grep arrow
 Downloading crates ...
  Downloaded const_fn v0.4.8
│   └── datafusion v4.0.0-SNAPSHOT (https://github.com/apache/arrow-datafusion.git?rev=7359e4b4df0836b8de970d551c6eeae22d1cc810#7359e4b4)
│       ├── arrow v4.1.0
│       │   ├── arrow v4.1.0 (*)
...
```

**Comment by alamb** [comment]
Draft until I update our parquet reading to handle the changes

File: parquet_file/src/storage.rs [hunk 2]
```diff
@@ -5,9 +5,9 @@ use arrow::{
     error::Result as ArrowResult,
     record_batch::RecordBatch,
 };
-use datafusion::physical_plan::{
-    common::SizedRecordBatchStream, parquet::RowGroupPredicateBuilder, RecordBatchStream,
-    SendableRecordBatchStream,
+use datafusion::{
```
Comment (alamb) [inline parquet_file/src/storage.rs:8]: The changes in this file are due to https://github.com/apache/arrow-datafusion/pull/370 (... that I caused...)

**Review by mkmik — APPROVED** [review]
Looks good

**Pull request merged**